### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by gmr**

### CODE REVIEW

File: sprockets/mixins/http/__init__.py [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+        :param str url: The URL for the request
+        :param str method: The HTTP request method, defaults to ``GET``
+        :param dict request_headers: Headers to include in the HTTP request
+        :param mixed body: The HTTP request body to send with the request
+        :param content_type: The mime type to use for requests & responses.
+            Defaults to ``application/msgpack``
+        :type content_type: :cls:`ietfparse.datastructures.ContentType` or str
+        :param bool follow_redirects: Follow HTTP redirects when received
+        :param float connect_timeout: Timeout for initial connection in
+            seconds, default 20 seconds
```
Comment (cinaglia) [inline sprockets/mixins/http/__init__.py:109]: It appears as though the default is actually 10 seconds.

File: sprockets/mixins/http/__init__.py [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+        :param float request_timeout:  Timeout for entire request in seconds,
+            default 20 seconds
```
Comment (cinaglia) [inline sprockets/mixins/http/__init__.py:111]: Same here, the default is appears to be 60 seconds.

**Comment by gmr** [comment]
The codecov errors are not indicating a problem with the PR, but the fact that the repo is new. There is 100% line coverage and 99% branch coverage with two weird edge cases that do have coverage but the coverage package is reporting as missing.

File: sprockets/mixins/http/__init__.py [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+        if hasattr(self, 'application'):
+            if self.application.settings.get('service') and \
+                    self.application.settings.get('version'):
+                return '{}/{}'.format(
+                    self.application.settings['service'],
+                    self.application.settings['version'])
+        return DEFAULT_USER_AGENT
+
+    def _http_resp_decode(self, value):
+        """Decode bytes to UTF-8 strings as a singe value, list, or dict.
```
Comment (cinaglia) [inline sprockets/mixins/http/__init__.py:248]: s/singe/single/

**Comment by cinaglia** [comment]
@gmr done reviewing. Only had minor typo comments. Otherwise, +1.

**Pull request merged**